### PR TITLE
[dv/pwrmgr] Add testpoint for main power glitch

### PR DESCRIPTION
**Pull request by matutem**
Signed-off-by: Guillermo Maturana <maturana@google.com>

### CODE REVIEW

File: hw/ip/pwrmgr/data/pwrmgr_testplan.hjson [hunk 1]
```diff
@@ -146,14 +146,28 @@
...
       milestone: V2
       tests: []
     }
+    {
+      name: main_power_glitch
+      desc: '''
+            Test that a main power glitch triggers a reset request.
+
+            **Stimulus**:
+            - Drop the AST input main_pok.
```
Comment (sriyerg) [inline hw/ip/pwrmgr/data/pwrmgr_testplan.hjson:162]: Do we need to cross this with various PWR FSM states? For example, what happens if `pok` drops while pwrmgr is progressing through low power entry or exit. 

@tjaychen

Comment (matutem) [reply hw/ip/pwrmgr/data/pwrmgr_testplan.hjson:162]: The logic depends on the state in some cases. Perhaps we should FPV this? Another way to test it is to flip pok while transitioning in and out of low power with main_pd_n on and off...

Comment (tjaychen) [reply hw/ip/pwrmgr/data/pwrmgr_testplan.hjson:162]: in theory, the same behavior applies as with all other reset requests.
if you are already in the middle of handling something, then the reset request is handled on the next cycle. The pok reset request is not handled any differently compared to other reset requests with the exception of always being enabled.

Comment (sriyerg) [reply hw/ip/pwrmgr/data/pwrmgr_testplan.hjson:162]: I suppose then our strategy should be to use `{rstreqs, pok}` wide input vector for the reset req tests?

**Review by sriyerg — APPROVED** [review]

File: hw/ip/pwrmgr/data/pwrmgr_testplan.hjson (hunk 1 above)
Comment (matutem) [reply hw/ip/pwrmgr/data/pwrmgr_testplan.hjson:162]: Good thinking. And some can be disabled, while others (like this) are always on.

File: hw/ip/pwrmgr/data/pwrmgr_testplan.hjson [hunk 2]
```diff
@@ -132,46 +131,26 @@
...
-            - Checks the output `pwr_rst_req.reset_cause` matches a reset
-              cause.
-            - Checks the output `pwr_rst_req.rstreqs` matches the enabled
-              resets.
-            '''
-      milestone: V2
-      tests: []
-    }
-    {
-      name: escalation_reset
```
Comment (sriyerg) [inline hw/ip/pwrmgr/data/pwrmgr_testplan.hjson:158]: The testplan needs to account for all features discreetly. Our test strategy enables us to write on test to verify 3 seemingly different features together, however it is not always clear to the reader (for example, a SW person helping us at the chip level) if a very specific feature, for example, reset due to escalation is already tested at the pwrmgr IP level. The first place they will look is the regression results web page, which will only list `reset` as the testpoint. 

I suggest keeping this (and the main power glitch) as separate testpoints, but have them point the the same implemented test with the `tests` key. It should probably also be ok to have a reduced description for those:
```
    {
      name: escalation_reset
      desc: '''Test the escalation reset.

            This is covered by the `reset` testpoint above. Cross-referenced for 
            completeness. 
            '''
      milestone: V2
      tests: []
    }
```

Comment (matutem) [reply hw/ip/pwrmgr/data/pwrmgr_testplan.hjson:158]: Good point. I will send another PR straightening this up.

**Pull request merged**